src/main/java/seedu/address/model/internship/ApplicationStatus.java

```diff
@@ -12,7 +12,7 @@
  * Represents an Internship's application status in Flagship.
  * Guarantees: immutable; is valid as declared in {@link #isValidApplicationStatus(String)}
  */
-public class ApplicationStatus {
+public class ApplicationStatus implements Comparable<ApplicationStatus> {
     public static final String MESSAGE_CONSTRAINTS =
```

Comment:
Nice! By implementing the Comparable interface, we ensure that all of these fields can be sorted. Thanks javin